File: Two-Pointer/maxArea.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        low = 0
        high = n - 1
        result = 0

        while low < high :
            result = max(result,min(height[low],height[high]) * (high - low))

            if height[low] <= height[high]: low += 1
            else : high -= 1
        
        return result

File: Two-Pointer/maxArea.js
function maxArea(height) {
  let result = 0;
  let i = 0;
  let j = height.length - 1;

  while (i < j) {
    const temp = Math.min(height[i], height[j]);

    result = Math.max(result, temp * (j - i));

    if (height[i] < height[j]) i++;
    else j--;
  }

  return result;
}
